contracts/SafeMinion.sol

```diff
@@ -0,0 +1,658 @@
+    }
+
+    function crossWithdraw(
+        address target,
+        address token,
+        uint256 amount,
+        bool transfer
+    ) external memberOnly {
+        // @Dev - Target needs to have a withdrawBalance functions
+        IMOLOCH(target).withdrawBalance(token, amount);
+
+        // Transfers token into DAO.
+        if (transfer) {
+            bool whitelisted = moloch.tokenWhitelist(token);
+            require(whitelisted, ERROR_NOT_WL);
+            require(
+                IERC20(token).transfer(address(moloch), amount),
+                ERROR_TX_FAIL
+            );
+        }
+
+        emit CrossWithdraw(target, token, amount);
+    }
+
+    //  -- Proposal Functions --
+    function proposeAction(
+        bytes memory transactions,
+        address withdrawToken,
+        uint256 withdrawAmount,
+        string calldata details,
+        bool memberOrModule
+    ) external memberOrModuleOnly returns (uint256) {
+        uint256 proposalId = moloch.submitProposal(
+            address(executor),
+            0,
+            0,
+            0,
+            molochDepositToken,
+            withdrawAmount,
+            withdrawToken,
+            details
+        );
+
+        saveAction(
+            proposalId,
+            transactions,
+            withdrawToken,
+            withdrawAmount,
+            memberOrModule
+        );
+
+        return proposalId;
+    }
+
+    function deleteAction(uint256 _proposalId)
+        external
+        executorOnly
+        returns (bool)
+    {
+        // check action exists
+        require(actions[_proposalId].proposer != address(0), ERROR_NO_ACTION);
+        delete actions[_proposalId];
+        emit ActionDeleted(_proposalId);
+        return true;
+    }
+
+    function saveAction(
+        uint256 proposalId,
+        bytes memory transactions,
+        address withdrawToken,
+        uint256 withdrawAmount,
+        bool memberOrModule
+    ) internal {
+        bytes32 id = hashOperation(transactions);
+        Action memory action = Action({
+            id: id,
+            proposer: msg.sender,
+            executed: false,
+            token: withdrawToken,
+            amount: withdrawAmount,
+            moloch: address(moloch),
+            memberOrModule: memberOrModule
+        });
+        actions[proposalId] = action;
+        emit ProposeNewAction(
+            id,
+            proposalId,
+            withdrawToken,
+            withdrawAmount,
+            address(moloch),
+            memberOrModule,
+            transactions
+        );
+    }
+
+    function executeAction(uint256 proposalId, bytes memory transactions)
+        external
+        returns (bool)
+    {
+        Action memory action = actions[proposalId];
+        require(!action.executed, ERROR_EXECUTED);
+        // Mark executed before doing any external stuff
+        actions[proposalId].executed = true;
+
+        if (action.memberOrModule) {
```

Comment:
Maybe I'm reading this wrong, but this logic appears to be duplicated from `modifier memberOrModule`, is it necessary?